libkbfs/fetch_decider.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"sync"
+	"time"
+
+	"github.com/keybase/client/go/logger"
+	"golang.org/x/net/context"
+)
+
+// fetchDecider is a struct that helps avoid having too frequent calls
+// into a remote server.
+type fetchDecider struct {
+	log     logger.Logger
+	fetcher func(ctx context.Context) error
+	tagKey  interface{}
+	tagName string
+	clock   clockGetter
```

Comment:
I don't really follow.  `clockGetter` is an interface, not a struct, so it can't really be embedded. Do you want something like this?

```go
type fetchDecider struct {
Clock

...
}
```

That would mean we couldn't swap out the clock at runtime, which defeats the purpose of passing in the getter.

But I might be misunderstanding what you're asking.